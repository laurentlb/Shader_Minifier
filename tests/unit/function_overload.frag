int sum0() { return 0; }

int sum1(int x) { return sum0() + x; }

int sum2(int x, int y) { return sum1(x) + y; }

int sum3(int x, int y, int z) { return sum2(x, y) + z; }

int sum4(int x, int y, int z, int w) { return sum3(x, y, z) + w; }

int mult1(int x) { return x; }

int mult2(int x, int y) { return mult1(x) * y; }

int mult3(int x, int y, int z) { return mult2(x, y) * z; }
