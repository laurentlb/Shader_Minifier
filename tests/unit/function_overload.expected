// Generated with  (https://github.com/laurentlb/Shader_Minifier/)
#ifndef FUNCTION_OVERLOAD_EXPECTED_
# define FUNCTION_OVERLOAD_EXPECTED_

const char *function_overload_frag =
 "int H()"
 "{"
   "return 0;"
 "}"
 "int H(int G)"
 "{"
   "return H()+G;"
 "}"
 "int H(int i,int G)"
 "{"
   "return H(i)+G;"
 "}"
 "int H(int i,int G,int E)"
 "{"
   "return H(i,G)+E;"
 "}"
 "int H(int i,int G,int E,int D)"
 "{"
   "return H(i,G,E)+D;"
 "}"
 "int G(int G)"
 "{"
   "return G;"
 "}"
 "int G(int i,int H)"
 "{"
   "return G(i)*H;"
 "}"
 "int G(int i,int H,int E)"
 "{"
   "return G(i,H)*E;"
 "}";

#endif // FUNCTION_OVERLOAD_EXPECTED_
